### PR TITLE
 Added samples for Poisson and OGD regression

### PR DESCRIPTION
**Pull request by shmoradims**
Related to #2522

### CODE REVIEW

File: docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/Regression/OnlineGradientDescent.cs [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+                Console.WriteLine($"Label: {p.Label:F3}, Prediction: {p.Score:F3}");
+
+            // Expected output:
+            //   OGD doesn't work: https://github.com/dotnet/machinelearning/issues/2425
+
+            // Evaluate the overall metrics
+            var metrics = mlContext.Regression.Evaluate(transformedTestData);
+            SamplesUtils.ConsoleUtils.PrintMetrics(metrics);
+            
+
```
Comment (rogancarr) [inline docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/Regression/OnlineGradientDescent.cs]: Spaces

File: docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/Regression/OnlineGradientDescent.cs [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+            // Convert IDataView object to a list.
+            var predictions = mlContext.Data.CreateEnumerable<Prediction>(transformedTestData, reuseRowObject: false).ToList();
+
+            // Look at 5 predictions
+            foreach (var p in predictions.Take(5))
+                Console.WriteLine($"Label: {p.Label:F3}, Prediction: {p.Score:F3}");
+
+            // Expected output:
+            //   OGD doesn't work: https://github.com/dotnet/machinelearning/issues/2425
```
Comment (rogancarr) [inline docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/Regression/OnlineGradientDescent.cs]: Wouldn't it be better to not put this on the official docs page?

Comment (rogancarr) [reply docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/Regression/OnlineGradientDescent.cs]: Also, the issue says that it doesn't have baselines, not that it doesn't work. Can you update the issue?

---
In reply to: [268338288](https://github.com/dotnet/machinelearning/pull/3067#discussion_r268338288) [](ancestors = 268338288)

Comment (sfilipi) [reply docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/Regression/OnlineGradientDescent.cs]: OGD works, there are 5 tests about it running regularly. 



---
In reply to: [268338288](https://github.com/dotnet/machinelearning/pull/3067#discussion_r268338288) [](ancestors = 268338288)

Comment (rogancarr) [reply docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/Regression/OnlineGradientDescent.cs]: How about we put the output in and add a `// TODO #ISSUE_NUMBER: Explanation` above the output.

---
In reply to: [268347907](https://github.com/dotnet/machinelearning/pull/3067#discussion_r268347907) [](ancestors = 268347907,268338288)

**Pull request merged**